### PR TITLE
csv, ctypes, configparser explanations

### PR DESCRIPTION
**Pull request by hatal175**
explained/fixed stubtest exceptions for these.

Worth mentioning that Array.__iter__ is there because of this issue:
https://github.com/python/mypy/issues/2220

### CODE REVIEW

File: stdlib/ctypes/__init__.pyi [hunk 1]
```diff
@@ -36,15 +36,25 @@ class CDLL(object):
...
-    def __init__(
-        self,
-        name: Optional[str],
-        mode: int = ...,
-        handle: Optional[int] = ...,
-        use_errno: bool = ...,
-        use_last_error: bool = ...,
-        winmode: Optional[int] = ...,
-    ) -> None: ...
+    if sys.version_info < (3, 8):
```
Comment (JelleZijlstra) [inline stdlib/ctypes/__init__.pyi:39]: Could you put the 3.8+ branch first? That's the general convention and makes the code easier to follow.

File: tests/stubtest_whitelists/py3_common.txt [hunk 2]
```diff
@@ -120,24 +120,24 @@ collections.abc.Generator.gi_yieldfrom
...
-ctypes.memset
-ctypes.pointer
-ctypes.string_at
-ctypes.wstring_at
+ctypes.Array.__iter__  # mypy doesn't support using __getitem__ instead of __iter__ so this is here https://github.com/python/mypy/issues/2220
+ctypes.CDLL._FuncPtr  # None at class level but initialized in __init__ to this value
+ctypes.memmove  # CFunctionType
+ctypes.memset  # CFunctionType
+ctypes.pointer  # imported C function
+ctypes.string_at  # docstring argument name is wrong
```
Comment (JelleZijlstra) [inline tests/stubtest_whitelists/py3_common.txt:139]: Might want to report this as a CPython bug

File: stdlib/ctypes/__init__.pyi [hunk 3]
```diff
@@ -36,15 +36,25 @@ class CDLL(object):
...
+    else:
+        def __init__(
+            self,
+            name: Optional[str],
+            mode: int = ...,
+            handle: Optional[int] = ...,
+            use_errno: bool = ...,
+            use_last_error: bool = ...,
+            winmode: Optional[int] = ...,
+        ) -> None: ...
```
Comment (JelleZijlstra) [inline stdlib/ctypes/__init__.pyi:57]:
```suggestion
    if sys.version_info >= (3, 8):
        def __init__(
            self,
            name: Optional[str],
            mode: int = ...,
            handle: Optional[int] = ...,
            use_errno: bool = ...,
            use_last_error: bool = ...,
            winmode: Optional[int] = ...,
        ) -> None: ...
    else:
        def __init__(
            self,
            name: Optional[str],
            mode: int = ...,
            handle: Optional[int] = ...,
            use_errno: bool = ...,
            use_last_error: bool = ...,
        ) -> None: ...
```

**Review by JelleZijlstra — COMMENTED** [review]
Thanks for all your work cleaning up the whitelists!

**Comment by hatal175** [comment]
Opened python bug:
https://bugs.python.org/issue43803

**Pull request merged**